dashboard/utils.py

```diff
@@ -324,6 +333,70 @@ def create_node_with_children(node_brother, notified_by, brothers_notified):
 def notifies(brother):
     return list(map(lambda node : node.brother, PhoneTreeNode.objects.filter(notified_by=brother)))
 
+
 def notified_by(brother):
     node = PhoneTreeNode.objects.filter(brother=brother)
     return node[0].notified_by if len(node) > 0 else None
+
+
+# zips a list of tuples: (event, attendance status)
+def create_attendance_list(events, excuses_pending, excuses_approved, brother):
+    """ zips together a list of tuples where the first element is each event and the second is the brother's
+        status regarding the event. If the event hasn't occurred, the status is blank, if it's not a mandatory
+        event it's 'not mandatory'
+
+    """
+    attendance = []
+    for event in events:
+        if int(event.date.strftime("%s")) > int(datetime.datetime.now().strftime("%s")):
+            attendance.append('')
+        elif not event.mandatory:
+            attendance.append('Not Mandatory')
+        else:
+            if event.attendees_brothers.filter(id=brother.id):
+                attendance.append('Attended')
+            elif event.pk in excuses_approved:
+                attendance.append('Excused')
+            elif event.pk in excuses_pending:
+                attendance.append('Pending')
+            else:
+                attendance.append('Unexcused')
+
+    return zip(events, attendance)
+
+
+def update_attendance_list(brother_form_list, brothers, event):
```

Comment:
what is this doing? update is not super descriptive.  at the very least documentation would be helpful.  im not sure what the intent of this method is